numpyro/util.py

```diff
@@ -136,8 +136,7 @@ def _body_fn(i, vals):
             collection = ops.index_update(collection, i, ravel_fn(val))
             return val, collection
 
-        _, collection = jit(lax.fori_loop, static_argnums=(2,))(0, n, _body_fn,
-                                                                (init_val, collection))
+        _, collection = fori_loop(0, n, _body_fn, (init_val, collection))
```

Comment:
Is this required for `pmap` or is this because there is no difference between jit and no-jit now - I remember there was a significant difference in some tests in `test_mcmc` earlier.

Reply:
This is required by `pmap`. I'll test if we really need 10s/8000 samples of jit(lax.fori_loop).

Reply:
It will be sad if we have to take a significant hit on performance just to be compatible with pmap. I would suggest adding a comment here and in `init_kernel` like `PERF: jitting the for loop may be faster on certain models / high number of samples`, so that we can easily track any perf issues we are leaving on the table and go back and undo it if needed (while benchmarking for instance).

Reply:
I don't know if which test makes it hits **significant**. Dropping jit in fori_collect costs about 1s for bernoulli, categorical tests (with 20000 samples). Testing on 10000 samples of hmm gives 106s vs 101s. But the costs are larger for warmup_update. I am seeing how much it costs.

Reply:
Hmm, actually I'm seeing different results of jit fori_loop and non-jit. So I'll spend more effort to debug what is the source of this problem.